File: Core/Domain/Entities/AnimePin.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;
using Core.Domain.IdentityEntities;
using Core.Enums;

namespace Core.Domain.Entities
{
    public class AnimePin
    {
        public Guid UserId { get; set; }
        public Guid AnimeId { get; set; }
        public string PinType { get; set; } = AnimePinTypeEnum.Watching.ToString(); //AnimePinTypeEnum
        public int? Grade { get; set; } //1-10
        public bool isFavorite { get; set; } = false;
        public ApplicationUser? User { get; set; }
        public Anime? Anime { get; set; }
    }
}

File: client-app/src/app/models/entities/animePin.ts
import { Anime } from "./anime"

export interface AnimePin {
    userId: string
    animeId: string
    pinType: string
    grade: number
    isFavorite: boolean
    addDate: string
    anime: Anime
  }

export interface AnimePinCreateValues {
  animeId: string
  pinType: string
  grade: number
  isFavorite: boolean
}
  